File: sdk/typescript-guest/templates/shipping.ts
import type * as Contract from '../generated/interfaces/examples-shipping-api.js';

// lsf-example-begin: capsule
export const api: typeof Contract = {
  quote(items, express) {
    if (items < 1 || items > 100) throw 'Choose between 1 and 100 items.';
    return (express ? 1200 : 500) + items * 75;
  },
};
// lsf-example-end: capsule

File: sdk/dotnet-guest/templates/shipping.cs
using ServiceWorld;
namespace ServiceWorld.wit.Exports.examples.shipping;

// lsf-example-begin: capsule
public class ApiExportsImpl : IApiExports
{
    public static Result<uint, string> Quote(uint items, bool express)
    {
        if (items < 1 || items > 100) return Result<uint, string>.Err("Choose between 1 and 100 items.");
        return Result<uint, string>.Ok((express ? 1200U : 500U) + items * 75U);
    }
}
// lsf-example-end: capsule
